graphql/util_test.go

```diff
@@ -351,3 +353,63 @@ func TestConcurrentlyBuildVersionsMatchingTasksMap(t *testing.T) {
 	assert.Equal(t, versionsMatchingTasksMap["v3"], true)
 
 }
+
+func TestHasAnnotationPermission(t *testing.T) {
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T){
+		"TrueWhenUserHasRequiredLevelAndIsNotPatchOwner": func(ctx context.Context, t *testing.T) {
+			task := restModel.APITask{ProjectId: utility.ToStringPtr("project_id_belonging_to_user"), Version: utility.ToStringPtr("random_version_id")}
+			hasAccess, err := hasAnnotationPermission(ctx, &task, evergreen.AnnotationsView.Value)
+			assert.NoError(t, err)
+			assert.True(t, hasAccess)
+		},
+		"FalseWhenUserDoesNotHaveRequiredLevelAndIsNotPatchOwner": func(ctx context.Context, t *testing.T) {
+			task := restModel.APITask{ProjectId: utility.ToStringPtr("project_id_belonging_to_user"), Version: utility.ToStringPtr("random_version_id")}
+			hasAccess, err := hasAnnotationPermission(ctx, &task, evergreen.AnnotationsModify.Value)
+			assert.NoError(t, err)
+			assert.False(t, hasAccess)
+		},
+		"TrueWhenUserIsPatchOwnerButDoesNotHaveRequiredLevel": func(ctx context.Context, t *testing.T) {
```

Comment:
I still think the test name is confusing because `"DoesNotHaveRequiredLevel"` sounds like the result should be false, but I'm not sure if that was intentional

I think the test names would be easier to be understand if they were like:
* `TrueForViewPermissionsWhenUserIsNotPatchOwner`
* `FalseForModifyPermissionsWhenUserIsNotPatchOwner`
* `TrueForViewPermissionsWhenUserIsPatchOwner`
or something along those lines!

Reply:
`DoesNotHaveRequiredLevel` refers to the `requiredLevel` function parameter.